File: 日次記録/2026年/4月/11日/task.py
import asyncio


async def task1():
    await asyncio.sleep(1)
    return "task1 done"


async def task2():
    await asyncio.sleep(2)
    return "task2 done"


async def main():
    results = await asyncio.gather(task1(), task2())
    print(results)


asyncio.run(main())

File: 日次記録/2026年/4月/11日/task.go
package main

import (
	"fmt"
	"time"
)

func task1(ch chan string) {
	time.Sleep(1 * time.Second)
	ch <- "task1 done"
}

func task2(ch chan string) {
	time.Sleep(2 * time.Second)
	ch <- "task2 done"
}

func task3(ch chan string) {
	time.Sleep(3 * time.Second)
	ch <- "task3 done"
}

func main() {
	ch := make(chan string)

	go task1(ch)
	go task2(ch)
	go task3(ch)

	result1 := <-ch
	result2 := <-ch
	result3 := <-ch

	fmt.Println(result1, result2, result3)
}
